lib/Backend/IRBuilder.cpp

```diff
@@ -6074,7 +6074,16 @@ IRBuilder::BuildProfiledCallI(Js::OpCode opcode, uint32 offset, Js::RegSlot retu
         newOpcode = opcode;
         Js::OpCodeUtil::ConvertNonCallOpToNonProfiled(newOpcode);
         Assert(newOpcode == Js::OpCode::NewScObject || newOpcode == Js::OpCode::NewScObjectSpread);
-        returnType = ValueType::GetObject(ObjectType::UninitializedObject);
+        if (newOpcode == Js::OpCode::NewScObjectSpread || !this->m_func->HasProfileInfo())
```

Comment:
I hadn't properly tested enabling it for NewScObjectSpread; the concern was that I would see similar performance degradations as with NewScObject, which would then require additional patches and expand the scope somewhat. I'm currently running tests on performance and functionality with profile data properly enabled for NewScObjectSpread return values.